src/qrcoder.jl

```diff
@@ -0,0 +1,112 @@
+# Add Functions to QRCode
+## Functions about polynomial
+import Base: %, ÷
+
+"""
+    %(f::Poly, g::Poly)
+
+Remainder of Euclidean division.
+"""
+%(f::Poly, g::Poly) = Poly((f + (f ÷ g) * g).coeff[1:(length(g) - 1)])
+
+"""
+    ÷(f::Poly, g::Poly)
+
+Quotient of Euclidean division.
+"""
+function ÷(f::Poly, g::Poly)
+    quodeg = length(f) - length(g) ## degree of the quotient polynomial
+    quodeg < 0 && return Poly([0])
+    g <<= quodeg
+    gn = lead(g) ## leading term of g(x)
+    quocoef = Vector{Int}(undef, quodeg + 1)
+    for i in 1:quodeg
+        quocoef[i] = divide(lead(f), gn)
+        f = init!(quocoef[i] * g + f)
+        tail!(g)
+    end
+    quocoef[end] = divide(lead(f), gn)
+    return Poly(reverse!(quocoef))
+end
+
+"""
+    divide(a::Int, b::Int)
+
+Division of intergers in GF(256).
+"""
+function divide(a::Int, b::Int)
+    b == 0 && throw(DivideError())
+    a == 0 && return 0
+    xa, xb = antilogtable[a], antilogtable[b]
+    ## use mod instead of % to avoid negative numbers
+    return logtable[mod(xa - xb, 255)]
+end
+
+
+## Functions about encoding & decoding
+"""
+    hamming_weight(x::Poly)
+
+Calculate the Hamming weight of a polynomial.
+"""
+hamming_weight(x::Poly) = hamming_weight(x.coeff)
+hamming_weight(x::Int) = hamming_weight(digits(x; base=2))
+hamming_weight(x::AbstractVector) = count(!=(0), x)
+
+
+"""
+    hamming_distance(x::Poly, y::Poly)
+
+Calculate the Hamming distance between two polynomials.
+"""
+hamming_distance(x::Poly, y::Poly) = hamming_distance(x.coeff, y.coeff)
+hamming_distance(x::Int, y::Int) = hamming_distance(digits(x; base=2), digits(y; base=2))
+hamming_distance(x::AbstractVector, y::AbstractVector) = hamming_weight(x - y)
```

Comment:
For fast-development purpose this is okay. But we'll need to move these to where it belongs: QRCode.jl -- technically, this is type piracy. https://docs.julialang.org/en/v1/manual/style-guide/#Avoid-type-piracy